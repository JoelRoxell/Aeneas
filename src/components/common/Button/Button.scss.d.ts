// This file is generated automatically
export const button: string;
export const primary: string;
export const secondary: string;
export const none: string;
export const activeLink: string;
export const inner: string;
export const left: string;
export const text: string;
export const icon: string;
