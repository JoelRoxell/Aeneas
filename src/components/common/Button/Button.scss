.button {
  cursor: pointer;
  height: rem(40);
  justify-content: center;
  align-items: center;
  display: flex;
  text-decoration: none;
  width: rem(220);
  font-size: rem(16);
  font-weight: bold;
  border-radius: rem(4);
  border: 0;
  color: #fff;
  fill: $primary;
  transition: background 0.3s;

  p {
    padding: 0;
    margin: 0;
  }

  &.primary {
    background-color: $primary;
  }

  &.secondary {
    background-color: $secondary;
  }

  &.none {
    background: none;
    box-shadow: none;
  }

  &.active-link {
    background: #f5f6f8;
    fill: $primary !important;
    transition: fill 0.3s;
  }

  &:active {
    box-shadow: inset 0 0 rem(8) rgba(0, 0, 0, 0.25);
  }

  &:hover {
    box-shadow: inset 0 0 0 rem(999) rgba(0, 0, 0, 0.125);
  }

  &:focus {
    box-shadow: none;
  }

  .inner {
    display: flex;
    align-items: center;
    width: fit-content;
    align-items: center;

    &.left {
      flex: 1;
      padding-left: 15%;
    }

    .text {
      flex: 1;
    }
  }

  .icon {
    $size: rem(20);
    margin-right: rem(18);
    height: $size;
    width: $size;
  }

  &.disabled {
    background-color: $light-gray;
  }
}
