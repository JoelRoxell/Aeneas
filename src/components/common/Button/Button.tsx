import * as React from 'react'
import * as classNames from 'classnames'

import { NavLink } from 'react-router-dom'

import * as style from './Button.scss'

interface ButtonProps {
  title: string
  to?: string
  type?: string
  className?: string
  disabled?: boolean
  icon?: string
  onClick?: (e: React.MouseEvent<HTMLButtonElement>) => any
  left?: boolean
}

Button.types = {
  SECONDARY: 'secondary',
  PRIMARY: 'primary'
}

function Button(props: ButtonProps) {
  let className

  switch (props.type) {
    case Button.types.PRIMARY:
      className = style.primary
      break
    case Button.types.SECONDARY:
      className = style.secondary
      break
    default:
      className = style.primary
  }

  return props.to ? (
    <NavLink
      activeClassName={style.activeLink}
      className={classNames(style.button, className, props.className)}
      to={props.disabled ? '' : props.to}
    >
      <div className={classNames(style.inner, { [style.left]: props.left })}>
        <p className={style.text}>{props.title}</p>
      </div>
    </NavLink>
  ) : (
    <button
      onClick={props.onClick}
      className={classNames(style.button, className, props.className)}
      disabled={props.disabled}
    >
      <p>{props.title}</p>
    </button>
  )
}

export default Button
