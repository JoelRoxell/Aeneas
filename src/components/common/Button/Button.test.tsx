import * as React from 'react'
import { shallow } from 'enzyme'
import Button from './Button'
// import {} from 'jest'

describe('Button', () => {
  test('Button onClick', () => {
    const cb = jest.fn(x => {})
    const button = shallow(<Button title="A clickable button" onClick={cb} />)

    // Interaction demo
    button.simulate('click').simulate('click')

    expect(button.text()).toEqual('A clickable button')
    expect(cb.mock.calls.length).toEqual(2)

    // Snapshot demo
    expect(button).toMatchSnapshot()
  })
})
