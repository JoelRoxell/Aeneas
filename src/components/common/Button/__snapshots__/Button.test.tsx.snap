// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button onClick 1`] = `
<button
  className="button primary"
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <p>
    A clickable button
  </p>
</button>
`;
